src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts

```diff
@@ -202,8 +222,37 @@ export class NotebookProviderInfoStore extends Disposable {
 					cellOptions = (options as INotebookEditorOptions | undefined)?.cellOptions;
 				}
 
-				const notebookOptions: INotebookEditorOptions = { ...options, cellOptions, viewState: undefined };
-				const editor = NotebookEditorInput.getOrCreate(this._instantiationService, notebookUri, preferredResource, notebookProviderInfo.id);
+				let notebookOptions: INotebookEditorOptions;
+
+				if (resource.scheme === Schemas.vscodeNotebookCellOutput) {
+					if (!data?.handle) {
+						throw new Error('Invalid cell handle');
+					}
+					const cellUri = data?.notebook.with({ fragment: (data.handle).toString() });
+
+					cellOptions = { resource: cellUri, options };
+
+					const cellIndex = await this._notebookEditorModelResolverService.resolve(notebookUri)
+						.then(model => model.object.notebook.cells.findIndex(cell => cell.uri.fragment === cellUri.fragment))
```

Comment:
I think you can use somethign as follows, this is better and more accurage.

```suggestion
						.then(model => model.object.notebook.cells.findIndex(cell => cell.handle === data?.handle))
```